Detector.cc

```
@@ -2088,14 +2088,30 @@ Detector::Detector(Settings *settings1, IceModel *icesurface, string setupfile)
 	    cout<<"start read elect chain"<<endl;
         if(settings1->CUSTOM_ELECTRONICS==0){
             //read the standard ARA electronics
-            cout<<"     Reading standard ARA electronics response"<<endl;
-             ReadElectChain("./data/ARA_Electronics_TotalGain_TwoFilters.txt", settings1);
-            //ReadElectChain("./data/ARA_Electronics_TotalGainPhase.txt", settings1);
-        }
+            if(settings1->DETECTOR_STATION==2 || settings1->DETECTOR_STATION==3){

```

Is it possible to make this if statement condition be something like `if(!settings1->DETECTOR_STATION != default_value)`? That way this won't need to be changed by hand as more stations' gain models are added.

---

A few lines earlier (L2075) Brian did if(settings1->DETECTOR_STATION > 0) for the noise to exclude the Testbed. I could do this? I was just concerned about being able to simulate A1 now with the standard gain model. This wouldn't be a problem when gain models are added for A1.

---

Maybe a solution would be to have `ReadElectChain` switch to default model (TotalGain_TwoFilters.csv I think?) if it find that `the_gain_filename` doesn't exist. And spit out an error message warning that it couldn't find the requested file and is falling back to the default.

---

Ok, I'll think what's the best way to do that. We'll also want it to fall back to default noise in that case, right?

---

Ya I think that's right. It shouldn't break things it should just throw an error message to warn the user.

---

I fixed it according to your first comment and used this instead: `if(settings1->DETECTOR_STATION > 0)`. To have the testbed use the default gain model. 

On your third comment, I think we should still have it throw an error message. That's how it's done for the noise too and the user may not notice that the noise/gain models were set to default unless they read the print file

---

Yes absolutely agree regarding still throwing the error message.